hybridcloud/backends/azure_postgresqlflexible.py

```diff
@@ -208,21 +210,51 @@ def compare():
         except ResourceNotFoundError:
             applied_allowed_extenions = []
         
+        should_restart = False
+
         if preload_extensions != applied_preload_extensions:
             self._logger.info(f"Updating list of extensions from {','.join(applied_preload_extensions)} to {','.join(extensions)}")
             # Update configuration
             poller = self._db_client.configurations.begin_put(self._resource_group, server_name, PRELOAD_PARAMETER, Configuration(value=",".join(extensions), source="user-override"))
             poller.result()
-            # Restart server
-            self._logger.info("Restarting server due to changed extensions preload configuration")
-            poller = self._db_client.servers.begin_restart(self._resource_group, server_name)
-            poller.result()
+            should_restart = True
         
         if extensions != applied_allowed_extenions:
             # Update configuration
             poller = self._db_client.configurations.begin_put(self._resource_group, server_name, EXTENSIONS_PARAMETER, Configuration(value=",".join(extensions), source="user-override"))
             poller.result()
+            should_restart = True
         
+        for parameter in self._db_client.configurations.list_by_server(self._resource_group, server_name):
```

Comment:
Comment what this for loop is doing, something along the lines of 

"iterate through the server properties that are currently set on the server" - then later add something to the comments like "comparing target server properties to current ones"